tests/post_training/pipelines/lm_weight_compression.py

```diff
@@ -0,0 +1,208 @@
+                self.fp32_model_dir, trust_remote_code=True, load_in_8bit=False, compile=False, stateful=False
+            )
+        self.model = self.model_hf.model
+
+    def prepare_preprocessor(self) -> None:
+        self.preprocessor = AutoTokenizer.from_pretrained(self.model_id)
+
+    def get_transform_calibration_fn(self):
+        def transform_fn(data):
+            tokenized_text = self.preprocessor(data["text"], return_tensors="np")
+            input_ids = tokenized_text["input_ids"]
+            attention_mask = tokenized_text["attention_mask"]
+
+            inputs = {}
+            inputs["input_ids"] = input_ids
+            inputs["attention_mask"] = tokenized_text["attention_mask"]
+            position_ids = np.cumsum(attention_mask, axis=1) - 1
+            position_ids[attention_mask == 0] = 1
+
+            # The magic forms KV cache as model inputs
+            batch_size = input_ids.shape[0]
+            for input_name in self.model_hf.key_value_input_names:
+                model_inputs = self.model.input(input_name)
+                shape = model_inputs.get_partial_shape()
+                shape[0] = batch_size
+                if shape[2].is_dynamic:
+                    shape[2] = 0
+                else:
+                    shape[1] = 0
+                inputs[input_name] = ov.Tensor(model_inputs.get_element_type(), shape.get_shape())
+
+            inputs["position_ids"] = position_ids
+            return inputs
+
+        return transform_fn
+
+    def prepare_calibration_dataset(self):
+        dataset = load_dataset("wikitext", "wikitext-2-v1", split="train")
+        dataset = dataset.filter(lambda example: len(example["text"]) > 80)
+        self.calibration_dataset = nncf.Dataset(dataset, self.get_transform_calibration_fn())
+
+    def cleanup_cache(self):
+        dir_with_cache = "model_cache"
+        dirs_to_remove = [self.output_model_dir / dir_with_cache, self.fp32_model_dir / dir_with_cache]
+        for dir_to_remove in dirs_to_remove:
+            if dir_to_remove.exists():
+                shutil.rmtree(dir_to_remove)
+
+    def compress(self) -> None:
+        if self.backend == BackendType.FP32:
+            return
+
+        print("Weight compression...")
+        start_time = time.perf_counter()
+        self.run_info.compression_memory_usage = memory_usage(self._compress, max_usage=True)
+        self.run_info.time_compression = time.perf_counter() - start_time
+
+    def collect_data_from_stdout(self, stdout: str):
+        stats = WCTimeStats()
+        stats.fill(stdout)
+        self.run_info.stats_from_output = stats
+
+    def save_compressed_model(self) -> None:
+        if self.backend == BackendType.FP32:
+            return
+        ov.serialize(self.model, self.output_model_dir / self.OV_MODEL_NAME)
+        self.model_hf._save_config(self.output_model_dir)
+
+    def get_num_compressed(self) -> None:
+        pass
+
+    def run_bench(self) -> None:
+        pass
```

Comment:
Why do you skip run benchmark_app?

Reply:
because usually LM are evaluated not via benchmark_app, but ov.genai